### PR TITLE
Intensive care uat gh16609

### PR DESCRIPTION
**Pull request by riadmen**
https://github.com/metasfresh/metasfresh/issues/16609

### CODE REVIEW

File: backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/ddl/functions/Fresh_PriceList_Details_ToDate_Report.sql [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+                      ORDER BY bp_value,
+                               plv_name,
+                               productcategory,
+                               m_product_id);
+
+    END IF;
+
+
+    DROP TABLE temp_price_list_details;
+
```
Comment (cristinamghita) [inline backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/ddl/functions/Fresh_PriceList_Details_ToDate_Report.sql:115]: You do not need to drop the temp table.

Comment (cristinamghita) [reply backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/ddl/functions/Fresh_PriceList_Details_ToDate_Report.sql:115]: And the customer function as explained in private

**Review by cristinamghita — CHANGES_REQUESTED** [review]
check my comment

File: backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/ddl/functions/Fresh_PriceList_Details_ToDate_Report.sql [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+             INNER JOIN M_PriceList_Version plv ON plv.M_PriceList_Version_ID = pp.M_PriceList_Version_ID
+             INNER JOIN M_PriceList pl ON plv.m_pricelist_id = pl.m_pricelist_id AND ps.m_pricingsystem_id = pl.m_pricingsystem_id AND pl.issopricelist = 'Y'
+        AND plv.validfrom >= p_validfrom;
+
+
+    IF p_show_product_price_pi_flag = 'N' THEN
+        RETURN QUERY (SELECT DISTINCT d.*, tmp.name::text AS plv_name
+                      FROM temp_price_list_details tmp
+                               INNER JOIN report.fresh_PriceList_Details_Report(
+                              tmp.c_bpartner_id,
```
Comment (cristinamghita) [inline backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/ddl/functions/Fresh_PriceList_Details_ToDate_Report.sql:85]: I think the IF is not ok.
In The original report we have an UNION.
I think here shall be the same.

**Review by cristinamghita — CHANGES_REQUESTED** [review]
See my comment and remove *.japer files.
We do not commit jasper files!

File: backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/ddl/functions/Fresh_PriceList_Details_ToDate_Report.sql (hunk 2 above)
Comment (riadmen) [reply backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/ddl/functions/Fresh_PriceList_Details_ToDate_Report.sql:85]: - as talked the union is just for the jasper,
 - jasper files removed

**Review by cristinamghita — APPROVED** [review]
lgtm

**Pull request merged**